### PR TITLE
feat: Adds `Select` and `SelectField` components

### PR DESCRIPTION
**Pull request by renatamottam**
> cherry-pick from https://github.com/vtex/faststore/pull/1529

## What's the purpose of this pull request?

- [x] Moves `Select` component to @faststore/components 
- [x] Moves `Select` styles on @faststore/ui
- [x] Adds basic `Icon` styles on @faststore/ui
- [x] Adds `Select` documentation page on Docusaurus
- [x] Creates `Select Field` component on @faststore/components 
- [x] Creates `Select Field` styles on @faststore/ui
- [x] Adds `Select Field` documentation page on Docusaurus

## How it works?
`Select` atom component is composed by `default select html tag` + `Icon`
`Select Field` molecule component is composed by `Select` + `Label`

## How to test it?

run yarn start in app/docs folder and checks the `Select` and `SelectField` components or test it using the doc preview link:
- [Select](https://faststore-git-feat-select-components-2-fs-616-faststore.vercel.app/reference/ui/atoms/Select)
- [SelectField](https://faststore-git-feat-select-components-2-fs-616-faststore.vercel.app/reference/ui/molecules/SelectField)

### Starters Deploy Preview
See it on Next PR: https://github.com/vtex-sites/nextjs.store/pull/314

### CODE REVIEW

File: packages/ui/src/components/atoms/Select/Select.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs'
```
Comment (eduardoformiga) [inline packages/ui/src/components/atoms/Select/Select.stories.mdx:1]: Maybe I got it wrong but aren't we planning to remove the storybook from `faststore/ui`?

**Review by eduardoformiga — COMMENTED** [review]
LGTM! I Left some comments along the way.

File: packages/ui/src/components/atoms/Icon/styles.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[data-fs-icon] {
```
[the remark this answers is not in the document]
Comment (renatamottam) [reply packages/ui/src/components/atoms/Icon/styles.scss:1]: yep, we're always centering it, so a basic style could help with that

File: packages/ui/src/assets/CaretDown.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
```
[the remark this answers is not in the document]
Comment (renatamottam) [reply packages/ui/src/assets/CaretDown.tsx:1]: the `CaretDown`? We use it on `Select`

File: apps/docs/src/css/custom.css [hunk 4]
```diff
@@ -176,7 +176,9 @@
  }
 
  hr {
-   @apply h-[0.5px] my-5 border-border !important;
+   @apply h-[1px] border-border !important;
```
Comment (hellofanny) [inline apps/docs/src/css/custom.css:179]: Do we need this? 👁️

File: apps/docs/docs/reference/ui/molecules/SelectField.mdx [hunk 5]
```diff
@@ -0,0 +1,98 @@
+---
+description: SelectField grouped a Select input and its corresponding Label.
+sidebar_custom_props:
+  image: https://vtexhelp.vtexassets.com/assets/docs/src/Select___8842f3b6522c3e8db0a887b3f216a6ab.png
```
Comment (hellofanny) [inline apps/docs/docs/reference/ui/molecules/SelectField.mdx:4]: Let's tag @carolinamenezes so we can remember to update this image later 🙈

File: packages/ui/src/components/molecules/SelectField/styles.scss [hunk 6]
```diff
@@ -0,0 +1,21 @@
+[data-fs-select-field] {
+  // --------------------------------------------------------
+  // Design Tokens for Select Field
+  // --------------------------------------------------------
+
+  // Label
+  --fs-select-field-label-color                 : var(--fs-color-text-light);
+  --fs-select-field-label-margin-right          : var(--fs-spacing-1);
```
Comment (hellofanny) [inline packages/ui/src/components/molecules/SelectField/styles.scss:8]:
```suggestion
  --fs-select-field-label-margin-right          : var(--fs-spacing-1);
  --fs-select-field-label-color                 : var(--fs-color-text-light);
```

File: packages/ui/src/components/atoms/Select/styles.scss [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+  --fs-select-transition-timing           : var(--fs-transition-timing);
+  --fs-select-transition-property         : var(--fs-transition-property);
+  --fs-select-transition-function         : var(--fs-transition-function);
+
+  // Icon
+  --fs-select-icon-color                  : var(--fs-color-link);
+  --fs-select-icon-position-right         : var(--fs-spacing-1);
+  --fs-select-icon-width                  : var(--fs-spacing-3);
+  --fs-select-icon-height                 : var(--fs-select-icon-width);
```
Comment (hellofanny) [inline packages/ui/src/components/atoms/Select/styles.scss:27]:
```suggestion
  --fs-select-icon-position-right         : var(--fs-spacing-1);
  --fs-select-icon-width                  : var(--fs-spacing-3);
  --fs-select-icon-height                 : var(--fs-select-icon-width);
  --fs-select-icon-color                  : var(--fs-color-link);
```

Comment (hellofanny) [reply packages/ui/src/components/atoms/Select/styles.scss:27]: I'm not sure. Are we following the default order for the tokens too? 👀

**Review by hellofanny — APPROVED** [review]
Well done renatinha!!!  🌟

File: apps/docs/src/css/custom.css (hunk 4 above)
Comment (renatamottam) [reply apps/docs/src/css/custom.css:179]: it's just a visual improvement for the doc 😬

**Pull request merged**